core/src/avm2/e4x.rs

```diff
@@ -540,6 +540,18 @@ impl<'gc> E4XNode<'gc> {
         }
     }
 
+    pub fn has_complex_content(&self) -> bool {
+        match &self.0.read().kind {
+            E4XNodeKind::Element { children, .. } => children
+                .iter()
+                .any(|child| matches!(&*child.kind(), E4XNodeKind::Element { .. })),
+            E4XNodeKind::Text(_) | E4XNodeKind::CData(_) => false,
```

Comment:
Can you combine all of these match arms into one?

Reply:
`cargo fmt` makes that look kinda ugly IMO
```
            E4XNodeKind::Text(_)
            | E4XNodeKind::CData(_)
            | E4XNodeKind::Attribute(_)
            | E4XNodeKind::Comment(_)
            | E4XNodeKind::ProcessingInstruction(_) => false,
```
So given that, I think I'll leave it as-is. Definitely a good suggestion though!